Marlin/src/gcode/geometry/G92.cpp

```diff
@@ -64,11 +59,7 @@ void GcodeSuite::G92() {
         LOOP_XYZE(i) {
           if (parser.seenval(axis_codes[i])) {
             current_position[i] = parser.value_axis_units((AxisEnum)i);
-            #if IS_SCARA || !HAS_POSITION_SHIFT
-              if (i == E_AXIS) didE = true; else didXYZ = true;
-            #elif HAS_POSITION_SHIFT
-              if (i == E_AXIS) didE = true;
-            #endif
+            if (i == E_AXIS) sync_E = true; else sync_XYZ = true;
```

Comment:
Is this definitely applicable to `SCARA` and setups with `NO_WORKSPACE_OFFSETS`? I'd hate to lose important logic related to these situations from a patch that only considers a default cartesian config.